### ISSUE TITLE
Crash on changing something in the lobby

### ISSUE THREAD

**Issue by Voker57** [freeorion/freeorion#1336]
Happens quite often with different players when somebody changes something

Backtrace: 

```
#0  0x00007f254a1f34d1 in ModalListPicker::CurrentItem (this=0x55a56143cd80) at /home/voker57/Sources/freeorion/GG/src/DropDownList.cpp:233
#1  0x00007f254a1f30bb in ModalListPicker::Run (this=0x55a56143cd80) at /home/voker57/Sources/freeorion/GG/src/DropDownList.cpp:183
#2  0x00007f254a1f66af in GG::DropDownList::LClick (this=0x55a5612cf6b0, pt=..., mod_keys=...)
    at /home/voker57/Sources/freeorion/GG/src/DropDownList.cpp:811
#3  0x000055a55c57fb61 in CUIDropDownList::LClick (this=0x55a5612cf6b0, pt=..., mod_keys=...)
    at /home/voker57/Sources/freeorion/UI/CUIControls.cpp:840
#4  0x00007f254a39dae3 in GG::Wnd::HandleEvent (this=0x55a5612cf6b0, event=...) at /home/voker57/Sources/freeorion/GG/src/Wnd.cpp:1070
#5  0x00007f254a296e00 in GG::GUIImpl::HandleMouseButtonRelease (this=0x55a55e66a300, mouse_button=0, pos=..., curr_ticks=42469)
    at /home/voker57/Sources/freeorion/GG/src/GUI.cpp:523
#6  0x00007f254a29a269 in GG::GUI::HandleGGEvent (this=0x7ffee1801328, event=GG::GUI::LRELEASE, key=GG::GGK_UNKNOWN, key_code_point=0, 
    mod_keys=..., pos=..., rel=..., text=0x0) at /home/voker57/Sources/freeorion/GG/src/GUI.cpp:1145
#7  0x00007f254c6a1084 in GG::SDLGUI::HandleSystemEvents (this=0x7ffee1801328) at /home/voker57/Sources/freeorion/GG/src/SDL/SDLGUI.cpp:801
#8  0x000055a55c3a6064 in HumanClientApp::HandleSystemEvents (this=0x7ffee1800860)
    at /home/voker57/Sources/freeorion/client/human/HumanClientApp.cpp:819
#9  0x00007f254a227eda in GG::ModalEventPump::operator() (this=0x7ffee18007c0) at /home/voker57/Sources/freeorion/GG/src/EventPump.cpp:113
#10 0x00007f254c6a13ee in GG::SDLGUI::Run (this=0x7ffee1801328) at /home/voker57/Sources/freeorion/GG/src/SDL/SDLGUI.cpp:883
#11 0x00007f254a299f23 in GG::GUI::operator() (this=0x7ffee1801328) at /home/voker57/Sources/freeorion/GG/src/GUI.cpp:1090
#12 0x00007f254c6a02d2 in GG::SDLGUI::operator() (this=0x7ffee1801328) at /home/voker57/Sources/freeorion/GG/src/SDL/SDLGUI.cpp:530
#13 0x000055a55c40d46d in mainSetupAndRun () at /home/voker57/Sources/freeorion/client/human/chmain.cpp:283
#14 0x000055a55c40a199 in main (argc=3, argv=0x7ffee1801808) at /home/voker57/Sources/freeorion/client/human/chmain.cpp:100
```

**Comment by l29ah** [comment]
same for me

**Comment by LGM-Doyle** [comment]
Thanks for the bug report.

I can't replicate this crash.

Could you provide some additional information?

What OS are you using?
What version/branch/commit are you playing?
Is this a recent change in the behavior of the program, or has this crash been happening for a while?
Does it only happen when multiple players have joined the lobby?

Do you have a simple step by step procedure to replicate the crash?

Once again thanks for the report.  I hope that you can flesh it out so that someone can replicate it.

**Comment by adrianbroher** [comment]
> Could you provide some additional information?

Why are you reiterating the questions in the issue template?  We should stop bothering and start closing this kind of issues as invalid.

**Comment by LGM-Doyle** [comment]
I'm hoping that by taking the time to ask the questions again I can shape these reports into reports that make better use of the reporter's time and mine.  They are taking the time to report it.  If they spend 5 more minutes, it will be something that someone can fix.

**Comment by adrianbroher** [comment]
> If they spend 5 more minutes, it will be something that someone can fix.

Like this already could have happened in the first place when opening the issue?

This isn't the first time and every damn time someone needed additionally query for relevant info.

https://github.com/freeorion/freeorion/issues?utf8=%E2%9C%93&q=author%3AVoker57

**Comment by Voker57** [comment]
The issue template is way too verbose and I don't feel like spending half an hour stating my graphics card model and surname of my deceased grandma (things of similar relevance to the issue) for every bug I report. If you want reporters to fill in all that information, script that does it for them, distributed with FreeOrion, would be helpful. 

Linux debian/sid
freeorion forked from 2627b390425789019 https://github.com/Voker57/freeorion/releases/tag/teamsmac-4. We don't appear to have any own patches for related code, I'll be able to test w/master tomorrow.

Bug have been happening for a while (several months), for debian, gentoo and windows users alike.
How to reproduce it: connect two players to server, start changing species for each player simultaneously. Soon FO will crash.

**Comment by adrianbroher** [comment]
> The issue template is way too verbose

Yeah, no.  https://github.com/freeorion/freeorion/blob/master/.github/ISSUE_TEMPLATE.md

> I don't feel like spending half an hour stating my graphics card model and surname of my deceased grandma (things of similar relevance to the issue) for every bug I report.

Thanks for confirming that you never read the issue template.

> You don't need to provide the information in section when they are not useful in
the context of this issue.  For example feature requests may not need a
FreeOrion version, but maybe depend on the operating system you use.  When in
doubt provide all information you have available.

**Comment by Vezzra** [comment]
@Voker57, the issue template is verbose the way it is because it tries to cover most of the use cases of a github issue. A feature request is different from a bug report, but AFAIK we can only set one template that github will use when someone opens an issue (so we need "one template to rule them all" 😉).

Which is why @adrianbroher (who came up with the idea and created the template) made sure to include several comments (like the one he quoted) in the template which explain how to use it, gives some examples for good/bad practice etc.

So no one expects you to fill in useless, irrelevant information, but the skeleton the template provides should help you determine which information we deem important in an issue, and giving as much and as detailed information as possible in your report helps us to react more efficiently, and also saves everyones time.

Most of the information you gave in response to @LGM-Doyle's inquiry could have been included in the original report, and wouldn't have taken you more than a couple of minutes to provide. We *do* appreciate that you take the time to provide your feedback (and it's very valuable to us since we don't get much feedback from people who do multiplayer games), but with your bug reports we frequently need to request the information needed to do anything about it, and that's certainly something that can (and should) be improved upon.

We're all volunteers here who do this stuff for fun in our free time, so all we ask is to help each other to make the best of the time each of us invests in this project. Filling out the issue template as best as you can is one way of doing that. 😃

**Comment by Voker57** [comment]
Just confirmed that it's present on 99fb4087f0216f178c05b, can assist with reproduction.

**Comment by LGM-Doyle** [comment]
@Voker57  could you provide the following?
- all the log files from both/all clients, `freeorion.log`, `freeoriond.log` files.  I'll need one `freeorion.log` from each human player.
- stack traces from the freeoriond, and both/all freeorion processes.  These will need to be from the same run, because we don't know which executable is crashing first.  

Thanks.

@Voker57 please continue submitting bug reports.  It is difficult as a develop and play multiplayer so these bugs are difficult to find and replicate.

If you want your bugs fixed, then they have to stand out from the crowd.  freeorion has 50 bugs today and I will only fix one.

Let me describe how I fix bugs to perhaps provide you context, for requests for more complete information in the original report.

When I have time for fixing bugs, I will grab an issue and try to replicate it on whatever build I have available.  If I can't replicate it, I will build a clean build of the version in the bug report, or master if there is no version in the bug report.  That takes a half hour.  If I still can't replicate it, I'll post online for some additional information.  I'll have to wait a day for that information, but I've already spent at least a half hour and accomplished nothing.  

Consequently, I avoid bugs that look difficult to replicate.  

If you want a bug fixed quickly in any project, not just freeorion, then its bug report should look easier to deal with than the other bugs.  In my own bug reports I try to include the maximal amount of information that is relevant to the problem, not the minimum information necessary to solve the problem.

**Comment by Voker57** [comment]
freeoriond and one of the human clients did not crash. freeorion stacktrace is included in the issue.

freeorion.log from non-crashed client: http://dump.bitcheese.net/files/xehovin/freeorion.log
freeorion.log from crashed client: https://dump.bitcheese.net/files/pafuhun/freeorion.log
freeoriond.log http://dump.bitcheese.net/files/zyhidas/freeoriond.log

ok, I will be more detailed in reports :-)

**Comment by LGM-Doyle** [comment]
Thanks.  

Do these log files go with the backtrace in the issue's first post?

**Comment by geoffthemedio** [comment]
>>The issue template is way too verbose
>
> Yeah, no. https://github.com/freeorion/freeorion/blob/master/.github/ISSUE_TEMPLATE.md

@adrianbroher to be fair to @Voker57 when one creates a new issue, there's a lot more to read in the box than shows in the page you linked. Most of it is comments, but it's a lot to read through, particularly for someone new.

(That said, "it crashed when I did something" is certainly not a useful bug report.)

````
<!--
Please fill in a a meaningful and concise issue title in the title field above.
It helps us to classify the issue even  before reading the body and handling the
issue a bit better.  Prefix the title with labels you find appropriate to
describe the issue like "[Bug]" in case of a bug, "[Feature]" for a feature
request.

Examples:
  * Great: "[Bug] Unable to select fleet in galaxy map"
  * Good: "[Bug] Unable to click fleet"
  * Not really okay: "Issue with fleets"
  * Horrible: "HELP!!!"
-->


### Environment
<!--
You don't need to provide the information in section when they are not useful in
the context of this issue.  For example feature requests may not need a
FreeOrion version, but maybe depend on the operating system you use.  When in
doubt provide all information you have available.

You can find the FreeOrion version number in the lower right corner of the game
main screen.

When stating the operating system also note the version of the operating system
so, `Windows 8 Pro` instead of `Windows` or `Mac OSX Mountain Lion` instead of
`OSX`.
-->

* **FreeOrion Version**:
* **Operating System**:
* **Graphic card used**:
* **Fetched as** <!-- delete lines below that don't apply -->
  * Binary release
  * Weekly development build
  * Compiled from source


### Description
<!--
Add a meaningful description of the bug you encountered or of the feature you
want to request.

If a screenshot or image helps to describe the issue content feel free to create
one.

Also attach the log files the game creates.  The log files The log files can be
found on
* Windows: %APPDATA%\FreeOrion
* MacOSX: ~/Library/"Application Support"/FreeOrion
* Linux: ${XDG_DATA_HOME:-~/.local/share}/freeorion
-->


### Expected Result
<!--
If this issue is not a bug you can remove this section.

When you enter an issue please add a description of what behaviour you would
expect from the game instead of the issue.
-->


### Steps to reproduce
<!--
If this issue is not a bug you can remove this section.

If the error only manifests itself after doing a certain number of actions
please add a list of steps to reproduce the bug.  If the error only occurs in
a certain game state please be sure to add a save game to the issue.
-->
* First step.
* Second step.
* Another step.
* Issue occurs.
````

**Comment by Vezzra** [comment]
@geoffthemedio, sure, but to be fair to @adrianbroher you only have to read all that once, after that you should know how to use the template efficiently. For a one-time read it's not really that much text.

And all those comments are there to help someone new to use it correctly. 😉

**Comment by geoffthemedio** [comment]
It long enough that my initial / immediate reaction to it would be to ignore most of it and just post what I think is relevant and useful.

I realize there's no good / perfect solution here, as you want to give some explanation and guidance for how to post a bug, but ideally the initial appearance would be much easier to look over and and get the idea of the scope of what's being requested.

Right now it could easily seem like way too much to bother reading through, particularly as it asks for things like graphics card or OS version that may be completely irrelevant to a bug, and someone might assume the whole thing is going to be questions like that that aren't relevant and thus not bother looking at the details after that.

**Comment by Voker57** [comment]
@LGM-Doyle it's from teamsmac branch but master produces the same (line numbers might be fuzzy)

**Comment by Vezzra** [comment]
Fixed by #1366.

**Comment by Vezzra** [comment]
@Voker57, just to be sure please check if the fix works for you and report back if it doesn't.
